diff --git a/ggml/src/ggml-vulkan/ggml-vulkan.cpp b/ggml/src/ggml-vulkan/ggml-vulkan.cpp
index 8108e94c1..000000000 100644
--- a/ggml/src/ggml-vulkan/ggml-vulkan.cpp
+++ b/ggml/src/ggml-vulkan/ggml-vulkan.cpp
@@ -3568,4 +3568,9 @@
     if (device->buffer_device_address) {
         const vk::BufferDeviceAddressInfo addressInfo(buf->buffer);
-        buf->bda_addr = device->device.getBufferAddress(addressInfo);
+        auto pfn_vkGetBufferDeviceAddress = reinterpret_cast<PFN_vkGetBufferDeviceAddress>(vkGetDeviceProcAddr(device->device, "vkGetBufferDeviceAddress"));
+        if (pfn_vkGetBufferDeviceAddress == nullptr) {
+            device->buffer_device_address = false;
+        } else {
+            buf->bda_addr = pfn_vkGetBufferDeviceAddress(device->device, reinterpret_cast<const VkBufferDeviceAddressInfo *>(&addressInfo));
+        }
     }
